enterprise/views.py

```diff
@@ -106,12 +122,54 @@ def get_note(provider, required):
                 " not be able to get any discounts from {provider}."
             ).format(provider=provider)
 
-    def get(self, request):
+    @staticmethod
+    def get_course_note(provider, course_name):
         """
-        Render a form to collect user input about data sharing consent.
+        Get a course-specific note for the form.
+
+        Arguments:
+            provider: The name of the linked EnterpriseCustomer
+            course_name: The name of the course in question
+        """
+        return _(
+            "Data sharing consent is required for courses provided by {provider}, including "
+            "{course_name}; if consent is not provided, you will be returned to the Dashboard."
+        ).format(provider=provider, course_name=course_name)
+
+    def get_course_specific_consent(self, request, course_id):
+        """
+        Render a form with course-specific information about data sharing consent.
+        """
+        try:
+            client = CourseApiClient()
+            course_details = client.get_course_details(course_id)
+        except Exception:
+            raise Http404
+        next_url = request.GET.get('next')
+
+        customer = get_object_or_404(EnterpriseCustomerUser, user_id=request.user.id).enterprise_customer
+        if not consent_necessary_for_course(request.user, course_id):
+            raise Exception
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+        course_name = course_details['name']
+        data = {
+            'platform_name': platform_name,
+            'data_sharing_consent': 'required',
+            "messages": {
+                "warning": self.get_course_warning(customer.name, course_name),
+                "note": self.get_course_note(customer.name, course_name),
+            },
+            'course_id': course_id,
+            'course_name': course_name,
+            'redirect_url': next_url,
+            'ec_name': customer.name,
+        }
+        return render_to_response('grant_data_sharing_permissions.html', data, request=request)
+
+    def get_account_consent(self, request):
+        """
+        Render a form to collect consent for account-wide data sharing.
         """
-        # Verify that all necessary resources are present
-        verify_edx_resources()
```

Comment:
So we're dropping this verification work from the `post`, but it is still present below in the `get` -- just wondering why we need it there but not here.